src/builder.rs

```diff
@@ -73,6 +75,168 @@ impl ConfigMapBuilder {
+        let mut sc = self
+            .security_context
+            .se_linux_options
+            .get_or_insert_with(SELinuxOptions::default);
+        sc.type_ = Some(type_.into());
+        self
+    }
+
+    pub fn se_linux_user(&mut self, user: impl Into<String>) -> &mut Self {
+        let mut sc = self
+            .security_context
+            .se_linux_options
+            .get_or_insert_with(SELinuxOptions::default);
+        sc.user = Some(user.into());
+        self
+    }
+
+    pub fn seccomp_profile_localhost(&mut self, profile: impl Into<String>) -> &mut Self {
+        let mut sc = self
+            .security_context
+            .seccomp_profile
+            .get_or_insert_with(SeccompProfile::default);
+        sc.localhost_profile = Some(profile.into());
+        self
+    }
+
+    pub fn seccomp_profile_type(&mut self, type_: impl Into<String>) -> &mut Self {
+        let mut sc = self
+            .security_context
+            .seccomp_profile
+            .get_or_insert_with(SeccompProfile::default);
+        sc.type_ = type_.into();
+        self
+    }
+
+    pub fn win_credential_spec(&mut self, spec: impl Into<String>) -> &mut Self {
+        let mut wo = self
+            .security_context
+            .windows_options
+            .get_or_insert_with(WindowsSecurityContextOptions::default);
+        wo.gmsa_credential_spec = Some(spec.into());
+        self
+    }
+
+    pub fn win_credential_spec_name(&mut self, name: impl Into<String>) -> &mut Self {
+        let mut wo = self
+            .security_context
+            .windows_options
+            .get_or_insert_with(WindowsSecurityContextOptions::default);
+        wo.gmsa_credential_spec_name = Some(name.into());
+        self
+    }
+
+    pub fn win_run_as_user_name(&mut self, name: impl Into<String>) -> &mut Self {
+        let mut wo = self
+            .security_context
+            .windows_options
+            .get_or_insert_with(WindowsSecurityContextOptions::default);
+        wo.run_as_user_name = Some(name.into());
+        self
+    }
+}
+
+/// Downward API capabilities available via `fieldRef`
+/// See: https://kubernetes.io/docs/tasks/inject-data-application/downward-api-volume-expose-pod-information/#capabilities-of-the-downward-api
+#[derive(Debug)]
+pub enum FieldPathEnvVar {
+    Name,
+    Namespace,
+    UID,
+    Labels(String),
+    Annotations(String),
+}
+
+impl fmt::Display for FieldPathEnvVar {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            FieldPathEnvVar::Name => write!(f, "metadata.name"),
+            FieldPathEnvVar::Namespace => write!(f, "metadata.namespace"),
+            FieldPathEnvVar::UID => write!(f, "metadata.uid"),
+            FieldPathEnvVar::Labels(name) => write!(f, "metadata.labels['{name}']"),
+            FieldPathEnvVar::Annotations(name) => write!(f, "metadata.annotations['{name}']"),
```

Comment:
```suggestion
            FieldPathEnvVar::Labels(name) => write!(f, "metadata.labels['{}']", name),
            FieldPathEnvVar::Annotations(name) => write!(f, "metadata.annotations['{}']", name),
```

`format!("{foo}")` requires rustc 1.58